patchset/0051-lavc-vaapi_hevc-support-420-422-444-12bit-enc-dec.patch

```diff
@@ -0,0 +1,418 @@
++            { 0, 8, 6, 4, 12, 7, 11, 7 },		/* U */
++            { 0, 8, 2, 4, 12, 7, 11, 3 },		/* V */
++            { 0, 8, 0, 4, 12, 7, 11, 1 },		/* A */
++        },
++        .flags = AV_PIX_FMT_FLAG_ALPHA | AV_PIX_FMT_FLAG_BE,
++    },
+     [AV_PIX_FMT_RGB24] = {
+         .name = "rgb24",
+         .nb_components = 3,
+@@ -2184,6 +2233,30 @@ static const AVPixFmtDescriptor av_pix_fmt_descriptors[AV_PIX_FMT_NB] = {
+         },
+         .flags = AV_PIX_FMT_FLAG_PLANAR | AV_PIX_FMT_FLAG_BE,
+     },
++    [AV_PIX_FMT_P012LE] = {
++        .name = "p012le",
++        .nb_components = 3,
++        .log2_chroma_w = 1,
++        .log2_chroma_h = 1,
++        .comp = {
++            { 0, 2, 0, 4, 12, 1, 11, 1 },        /* Y */
++            { 1, 4, 0, 4, 12, 3, 11, 1 },        /* U */
++            { 1, 4, 2, 4, 12, 3, 11, 3 },        /* V */
++        },
++        .flags = AV_PIX_FMT_FLAG_PLANAR,
++    },
++    [AV_PIX_FMT_P012BE] = {
++        .name = "p012be",
++        .nb_components = 3,
++        .log2_chroma_w = 1,
++        .log2_chroma_h = 1,
++        .comp = {
++            { 0, 2, 0, 4, 12, 1, 11, 1 },        /* Y */
++            { 1, 4, 0, 4, 12, 3, 11, 1 },        /* U */
++            { 1, 4, 2, 4, 12, 3, 11, 3 },        /* V */
++        },
++        .flags = AV_PIX_FMT_FLAG_PLANAR | AV_PIX_FMT_FLAG_BE,
++    },
+     [AV_PIX_FMT_P016LE] = {
+         .name = "p016le",
+         .nb_components = 3,
+diff --git a/libavutil/pixfmt.h b/libavutil/pixfmt.h
+index 1af1ec7cd5..cb55937cb3 100644
+--- a/libavutil/pixfmt.h
++++ b/libavutil/pixfmt.h
+@@ -297,6 +297,8 @@ enum AVPixelFormat {
+     AV_PIX_FMT_GRAY10BE,   ///<        Y        , 10bpp, big-endian
+     AV_PIX_FMT_GRAY10LE,   ///<        Y        , 10bpp, little-endian
+ 
++    AV_PIX_FMT_P012LE, ///< like NV12, with 12bpp per component, little-endian
++    AV_PIX_FMT_P012BE, ///< like NV12, with 12bpp per component, big-endian
+     AV_PIX_FMT_P016LE, ///< like NV12, with 16bpp per component, little-endian
+     AV_PIX_FMT_P016BE, ///< like NV12, with 16bpp per component, big-endian
+ 
+@@ -358,10 +360,16 @@ enum AVPixelFormat {
+     AV_PIX_FMT_Y210BE,    ///< packed YUV 4:2:2 like YUYV422, 20bpp, data in the high bits, big-endian
+     AV_PIX_FMT_Y210LE,    ///< packed YUV 4:2:2 like YUYV422, 20bpp, data in the high bits, little-endian
+ 
++    AV_PIX_FMT_Y212BE,    ///< packed YUV 4:2:2 like YUYV422, 24bpp, data in the high bits, big-endian
++    AV_PIX_FMT_Y212LE,    ///< packed YUV 4:2:2 like YUYV422, 24bpp, data in the high bits, little-endian
++
+     AV_PIX_FMT_0YUV,      ///< packed YUV 4:4:4, 32bpp,  X  Y Cb Cr, X=unused/undefined
+     AV_PIX_FMT_Y410LE,    ///< packed YUV 4:4:4, 32bpp, Cr  Y Cb  A, little-endian
+     AV_PIX_FMT_Y410BE,    ///< packed YUV 4:4:4, 32bpp, Cr  Y Cb  A, big-endian
+ 
++    AV_PIX_FMT_Y412LE,    ///< packed YUV 4:4:4, 36bpp, Cr  Y Cb  A, little-endian
++    AV_PIX_FMT_Y412BE,    ///< packed YUV 4:4:4, 36bpp, Cr  Y Cb  A, big-endian
```

Comment:
squash into an enum structure would lead to an ABI break, would be better to append at the end.
However since it's for internal usage, I'm ok with this (but needs to be adjusted while upstreaming).